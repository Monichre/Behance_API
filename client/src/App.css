body {
  font-family: "Open Sans";
  font-size: 0.75rem;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
ul {
  list-style: none;
  padding: 0;
}
a {
  cursor: pointer;
}
/* LOADER */
.loader_bg {
  position: fixed;
  top: 0; left: 0;
  height: 100%;
  width: 100%;
  z-index: 2000;
  -webkit-font-smoothing: antialiased;
  background: #edebea;
  
}

.loader,
.loader__figure {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  -o-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.loader {
  overflow: visible;
  padding-top: 2em;
  height: 0;
  width: 2em;
}

.loader__figure {
  height: 0;
  width: 0;
  box-sizing: border-box;
  border: 0 solid #FE665C;
  border-radius: 50%;
  -webkit-animation: loader-figure 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  -moz-animation: loader-figure 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  animation: loader-figure 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
}

.loader__label {
  float: left;
  margin-left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  -o-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0.5em 0 0 50%;
  font-size: 0.875em;
  letter-spacing: 0.1em;
  line-height: 1.5em;
  color: #8d3b3e;
  white-space: nowrap;
  -webkit-animation: loader-label 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  -moz-animation: loader-label 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
  animation: loader-label 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1);
}

@-webkit-keyframes loader-figure {
  0% {
    height: 0;
    width: 0;
    background-color: #FE665C;
  }
  29% {
    background-color: #FE665C;
  }
  30% {
    height: 2em;
    width: 2em;
    background-color: transparent;
    border-width: 1em;
    opacity: 1;
  }
  100% {
    height: 2em;
    width: 2em;
    border-width: 0;
    opacity: 0;
    background-color: transparent;
  }
}
@-moz-keyframes loader-figure {
  0% {
    height: 0;
    width: 0;
    background-color: #FE665C;
  }
  29% {
    background-color: #FE665C;
  }
  30% {
    height: 2em;
    width: 2em;
    background-color: transparent;
    border-width: 1em;
    opacity: 1;
  }
  100% {
    height: 2em;
    width: 2em;
    border-width: 0;
    opacity: 0;
    background-color: transparent;
  }
}
@keyframes loader-figure {
  0% {
    height: 0;
    width: 0;
    background-color: #FE665C;
  }
  29% {
    background-color: #FE665C;
  }
  30% {
    height: 2em;
    width: 2em;
    background-color: transparent;
    border-width: 1em;
    opacity: 1;
  }
  100% {
    height: 2em;
    width: 2em;
    border-width: 0;
    opacity: 0;
    background-color: transparent;
  }
}
@-webkit-keyframes loader-label {
  0% {
    opacity: 0.25;
  }
  30% {
    opacity: 1;
  }
  100% {
    opacity: 0.25;
  }
}
@-moz-keyframes loader-label {
  0% {
    opacity: 0.25;
  }
  30% {
    opacity: 1;
  }
  100% {
    opacity: 0.25;
  }
}
@keyframes loader-label {
  0% {
    opacity: 0.25;
  }
  30% {
    opacity: 1;
  }
  100% {
    opacity: 0.25;
  }
}

/* HOME */

#Home header {
  position: absolute;
  right: 0;
  height: 0;
}

/*  */

.account-box li:hover {
  background: #233140;
}

.account-box a {
  text-decoration: none;
  color: #FFF;
}

aside {
  background: #f5f7f7;
  width: 20%;
  height: 100vh;
  position: fixed;
  padding-top: 50px;
}

aside h1 {
  color: #5f6876;
  text-align: center;
  margin-top: 20px;
  font-size: 40px;
  font-size: 2.5rem;
}

aside ul {
  margin: 40px 0;
}


aside li.active {
  background: #f5f7f7;
}

aside li:hover {
  background: #f5f7f7;
}

aside a {
  color: #FFF;
  text-decoration: none;
  font-size: 12.5px;
  padding-left: 20px;
  font-weight: 400;
  text-transform: uppercase;
  font-size: 12.5px;
  font-size: 0.78125rem;
}

.dashboard {
  background: linear-gradient(45deg, #f4ebe5, #e3d4ca);
  padding: 150px 30px 0;
  width: 80%;
  float: right;
}

.ui.cards {
  width: 100%;
  margin: auto;
  justify-content: center;
}

.card .main_image {
  
  object-fit: contain;
  object-position: center;
  height: 200px;
  width: 100%;
}
.card .meta img {
  margin-right: 5px;
}


/* .info {
  padding: 25px;
}

.circle {
  width: 100px;
  height: 100px;
  display: table-cell;
  text-align: center;
  vertical-align: middle;
  border-radius: 50%;
  border: 10px solid #273646;
  color: #FFF;
}

.fa {
  color: #FFF;
  padding-right: 10px;
  font-size: 12.5px !important;
  font-size: 0.78125rem;
}

.fa.fa-caret-down {
  margin: 0 10px;
}

.fa.fa-bars {
  float: right;
  margin: 30px 20px;
  visibility: hidden;
  font-size: 20px !important;
  font-size: 1.25rem;
} */


/* SEARCH */

.search-wrapper {
  position: absolute;
  top: 30px;
  right: 80px;
}

.search-wrapper.active::after {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  content: '';
  background: rgba(255, 255, 255, .9);
  opacity: 1;
  z-index: 1000;
  transition: all .5s;
}

.search-content {
  position: absolute;
  display: none;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  flex-direction: row;
  justify-content: flex-start;
  z-index: 1000;
  padding: 150px 100px 0px;
}

.search-content h1 {
  font-size: 7em;
  position: absolute;
  top: 25px;
  left: 100px;
  color: rgba(236, 90, 98, 1);
}

.results-column {
  max-height: 600px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
}

.search-content .searchResult {
  padding: 0.75em;
  display: block;
  margin: 3px;
  cursor: pointer;
  border-radius: 5px;
  background: rgba(118, 117, 128, 0.05);
  border: 1px solid rgba(118, 117, 128, 0.08)
}



.search-wrapper.active .input-holder,
.search-wrapper.active .close {
  z-index: 1001;
}

.search-wrapper.active {}

.search-wrapper .input-holder {
  height: 70px;
  width: 70px;
  overflow: hidden;
  background: rgba(255, 255, 255, 0);
  border-radius: 6px;
  position: relative;
  transition: all 0.3s ease-in-out;
}

.search-wrapper.active .input-holder {
  width: 450px;
  border-radius: 50px;
  background: rgba(0, 0, 0, 0.5);
  transition: all .5s cubic-bezier(0.000, 0.105, 0.035, 1.570);
}

.search-wrapper .input-holder .search-input {
  width: 100%;
  height: 50px;
  padding: 0px 70px 0 20px;
  opacity: 0;
  position: absolute;
  top: 0px;
  left: 0px;
  background: transparent;
  box-sizing: border-box;
  border: none;
  outline: none;
  font-family: "Open Sans", Arial, Verdana;
  font-size: 16px;
  font-weight: 400;
  line-height: 20px;
  color: #FFF;
  transform: translate(0, 60px);
  transition: all .3s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  transition-delay: 0.3s;
}

.search-wrapper.active .input-holder .search-input {
  opacity: 1;
  transform: translate(0, 10px);
}

.search-wrapper .input-holder .search-icon {
  width: 70px;
  height: 70px;
  border: none;
  border-radius: 6px;
  background: #FFF;
  padding: 0px;
  outline: none;
  position: relative;
  z-index: 2;
  float: right;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.search-wrapper.active .input-holder .search-icon {
  width: 50px;
  height: 50px;
  margin: 10px;
  border-radius: 30px;
}

.search-wrapper .input-holder .search-icon span {
  width: 22px;
  height: 22px;
  display: inline-block;
  vertical-align: middle;
  position: relative;
  transform: rotate(45deg);
  transition: all .4s cubic-bezier(0.650, -0.600, 0.240, 1.650);
}

.search-wrapper.active .input-holder .search-icon span {
  transform: rotate(-45deg);
}

.search-wrapper .input-holder .search-icon span::before,
.search-wrapper .input-holder .search-icon span::after {
  position: absolute;
  content: '';
}

.search-wrapper .input-holder .search-icon span::before {
  width: 4px;
  height: 11px;
  left: 5px;
  top: 12px;
  border-radius: 2px;
  background: #FE5F55;
}

.search-wrapper .input-holder .search-icon span::after {
  width: 14px;
  height: 14px;
  left: 0px;
  top: 0px;
  border-radius: 16px;
  border: 4px solid #FE5F55;
}

.search-wrapper .close {
  position: absolute;
  z-index: 1;
  top: 24px;
  right: 20px;
  width: 25px;
  height: 25px;
  cursor: pointer;
  transform: rotate(-180deg);
  transition: all .3s cubic-bezier(0.285, -0.450, 0.935, 0.110);
  transition-delay: 0.2s;
}

.search-wrapper.active .close {
  right: -50px;
  transform: rotate(45deg);
  transition: all .6s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  transition-delay: 0.5s;
}

.search-wrapper .close::before,
.search-wrapper .close::after {
  position: absolute;
  content: '';
  background: #FE5F55;
  border-radius: 2px;
}

.search-wrapper .close::before {
  width: 5px;
  height: 25px;
  left: 10px;
  top: 0px;
}

.search-wrapper .close::after {
  width: 25px;
  height: 5px;
  left: 0px;
  top: 10px;
}

.profile_container {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: linear-gradient(45deg, #f4ebe5, #e3d4ca);
  color: #c0c3c9;
  line-height: 0.875rem;
  padding: 100px;
}  

.profile {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: #fff;
  width: 100%;
  height: 42.5rem;
  border-radius: 0.4375rem;
  -webkit-box-shadow: 1.375rem 2rem 2.125rem rgba(0, 0, 0, 0.06);
  box-shadow: 1.375rem 2rem 2.125rem rgba(0, 0, 0, 0.06);
  overflow: hidden;
}

.profile-sidebar {
  -ms-flex-preferred-size: 14.0625rem;
  flex-basis: 14.0625rem;
  width: 14.0625rem;
  min-width: 14.0625rem;
  border-right: 0.0625rem solid #f2f2f2;
}
.profile-preview__header .profile-nav {
  display: flex;
}
.profile-preview__header .profile-nav div {
  padding: 5px;
}
.profile-nav h1, .profile-preview__header h1 {
  color: #5f6876;
  letter-spacing: 0.1375rem;
  font-size: 0.8rem;
  padding: 1.3125rem 2.25rem;
}
.profile-preview__header .profile-nav h1, .profile-preview__header h1 {
  padding: 1.4rem 0 0;
}
.profile-sidebar__brand {
  margin: 0 0 1.4375rem;
  padding: 1.4375rem 1.625rem;
}
/* .list.followers > .item, .list.following > .item {
  margin: 0 5px;
  font-size: 0.8rem;
} */
/* .ui.horizontal.list.followers > .item span, .ui.horizontal.list.following > .item span {
  margin-left: 23px;
} */
.list.followers > .item img, .list.following > .item img {
  display: inline-block;
  height: 20px;
  width: 20px;

}


.profile-nav__item {
  background: 0;
}

.profile-nav__item a, aside ul li {
  display: block;
  padding: 1.3125rem 2.25rem;
  color: inherit;
  text-decoration: none;
}

.profile-nav__item a img {
  margin: 0 1.5rem 0 0;
}
.profile_social_links {
  position: absolute;
  top: 0;
  right: 10px;
}
.profile-nav__item--selected {
  background: #f5f7f7;
  color: #5f6876;
}

.profile-content {
  width: 100%;
  border-right: 0.0625rem solid #f2f2f2;
}

.profile-content__panel {
  display: none;
}

.profile-content__panel--active {
  display: block;
}

.profile-list__item {
  padding: 2.69rem;  
  border-bottom: 0.0625rem solid #f2f2f2;
  cursor: pointer;
}

.profile-list__item h2 {
  
  color: #5f6876;
  letter-spacing: 0.15rem;
}

.profile-list__item span {
  margin: 0 0 1.5rem;
  font-size: 0.625rem;
  letter-spacing: 0.13625rem;
}

.profile-list__item--active {
  background: #f5f7f7;
  position: relative;
}
.profile-list__item--active .fields, .stats_list {
  margin: 0 0 1.6875rem;
}
.ui.horizontal.list.stats_list {
  text-align: center;
width: 100%;
}
.ui.horizontal.list.stats_list > .item.stat {
  text-align: center;
  font-size: 0.8rem;
}
.stat {
  font-weight: 600;
  color: #5f6876;
}
.profile-list__avatar {
  display: inline-block;
  max-width: 1.9375rem;
  max-height: 1.9375rem;
  border-radius: 100%;
  overflow: hidden;
  margin: 0 0.6875rem 0 0;
}

.profile-header {
  padding: 2.0625rem 2.5625rem;
  border-bottom: 0.0625rem solid #f2f2f2;
  color: #5f6876;
  letter-spacing: 0.15rem;
  text-align: center;
}
.profile-header h2 {
  color: #5f6876;
  letter-spacing: 0.1375rem;
  font-size: 0.625rem;
}


/* .profile-header__search input {
  outline: none;
  display: block;
  width: 100%;
  padding: 23px;
  border: 0;
  letter-spacing: 0.125rem;
}

.profile-header__new {
  min-width: 3.75rem;
  padding: 23px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  cursor: pointer;
} */

.profile-preview {
  background: url("") no-repeat;
  background-size: cover;
  width: 100%;
  height: 100%;
}
/* 
.profile-preview__panel {
  display: none;
} */

.profile-preview__panel--active {
  display: block;
}
.profile-preview__panel--active .item {
  font-size: 0.75rem;
  margin: 1.0625rem 0 0 0;
  color: #5f6876;
  font-size: 0.75rem;
}
.profile-preview__panel--active .stat span {
  font-weight: 600;
  color: #5f6876;
  

}

.profile-preview__header {
  padding: 0 1.8125rem 1.6875rem;
  color: #fff;
}

.profile-preview__header h2 {
  font-size: 0.875rem;
  line-height: 1.0625rem;
}

.profile-preview__header h3 {
  font-size: 2.25rem;
  line-height: 2.875rem;
}

.profile-preview__header h3 span {
  font-size: 1rem;
  line-height: 2.25rem;
  vertical-align: top;
}

.profile-preview__content {
  padding: 1.6875rem 1.8125rem;
}

.profile-preview__content h2 {
  color: #5f6876;
  letter-spacing: 0.1375rem;
}

.profile-preview__content h2 span {
  color: #c0c3c9;
  font-size: 0.625rem;
}

.profile-preview__content label {
  display: block;
  margin: 1.0625rem 0 0 0;
  color: #5f6876;
}


@media screen and (max-width: 1100px) {
  .profile_container {
    padding: 25px;
  }
  .profile-sidebar__brand {
    padding: 1.4375rem 1.625rem 0;
  }
  .following > .item, .followers > .item {
    margin-top: 0;
  }
  .ui.list>.item>img.image {
    height: 20px;
    width: 20px;
  }
  .profile-preview__header {
    padding: 0;
  }
}
@media screen and (max-width: 980px) {
  .profile-preview__header .profile-nav {
    flex-direction: column;
  }
  .work_experience {
    overflow-y: scroll;
  }
  .profile-header span {
    font-size: 0.625rem;
  }
}


@media screen and (max-width: 786px) {
  .profile {
    flex-direction: column; 
  }
  .profile-sidebar {
    width: 100%;
    display: flex;
  }
  .profile-nav {
    display: block;
  }
  .profile-nav ul {
    display: flex;
  }
}