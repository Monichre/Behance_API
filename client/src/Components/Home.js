import React, { Component } from 'react'
import Axios from 'axios'
import { Card, Icon, Image, List, Divider } from 'semantic-ui-react'
import { Search } from '../Partials/Search'
import { Constants } from '../constants.js'

export default class Home extends Component {
    constructor(props) {
        super(props)

        this.state = {
            searchActive: false,
            searchTerm: '',
            items: [],
            fields: [],
            searchResults: [],
            searchCompleted: false
        }
    }
    componentWillMount() {
        const _this = this
        Axios.get('/gallery').then((response) => {
            let items = response.data.gallery.splice(0, 20)
            let fields = response.data.field_data.splice(0, 20)
            _this.setState({
                items: items,
                fields: fields
            })
        }).catch((error) => {
            console.log(error)
        })
    }
    startSearch() {
        this.setState({
            searchActive: true
        })
    }
    runSearch() {
        const _this = this
        
        Axios.post('/search', {
            data: _this.state.searchTerm
        }).then((response) => {
            _this.setState({
                searchResults: response.data,
                searchCompleted: true

            }, () => {
                console.log(this.state)
            })
        }).catch((error) => {
            console.log(error)
        })
    }
    closeSearch(e) {
        e.preventDefault()
        this.setState({
            searchActive: false,
            searchCompleted: false
        })
    }
    handleSearchInput(e) {
        e.preventDefault()
        this.setState({
            searchTerm: e.target.value
        }, ()=> {
            console.log(this.state)
        })
    }
    handleKeyPress(e) {
       if (e.which === 13) {
        console.log("you pressed enter")
        this.runSearch()
       }
    }

    render() {
        let search_results_style
        if (this.state.searchCompleted) {
            search_results_style = {
                display: 'flex'
            }
        } else {
            search_results_style = {
                display: 'none'
            }
        }
        return (
            <div id="Home">
                <header>
                    <i className="fa fa-bars"></i>
                    <div className="menu">
                        <div className={`search-wrapper ${this.state.searchActive ? 'active' : ''}`}>
                            <div className="input-holder">
                                <input type="text" className="search-input" value={this.state.searchTerm} onChange={this.handleSearchInput.bind(this)} placeholder="Type to search" onKeyDown={this.handleKeyPress.bind(this)}/>
                                <button className="search-icon" onClick={this.startSearch.bind(this)}><span></span></button>
                            </div>
                            <span className="close" onClick={this.closeSearch.bind(this)}></span>
                        </div>
                    </div>
                </header>

                <aside>
                    <h1>Dealer Inspire + Behance</h1>
                    <ul>
                        {this.state.fields.map((field) => (
                            <li>{field.name}</li>
                        ))}
                    </ul>
                </aside>

                <div className="dashboard">
                    <Search style={search_results_style} searchResults={this.state.searchResults} />
                    <Card.Group>
                        {this.state.items.map((item) => (
                            <Card>
                                <Image className="main_image" src={item.project_covers.length > 0 ? item.project_covers[0].url : ''} />
                                <Card.Content>
                                    <Card.Header>{item.title}</Card.Header>
                                    <Divider />
                                    <Card.Meta><Image src={item.multiple_owners === false ? item.owners[0].images['50'] : ''} circular={true} size='mini' /> {item.multiple_owners === false ? item.owners[0].first_name : ''} {item.multiple_owners === false ? item.owners[0].last_name : ''}: {item.multiple_owners === false ? item.owners[0].city : ''}, {item.multiple_owners === false ? item.owners[0].country : ''}</Card.Meta>
                                </Card.Content>
                                <Card.Content extra>
                                    <List horizontal>
                                        <List.Item icon='user' content={`Followers: ${item.follower_count}`} />
                                        <List.Item icon='tasks' content={`Projects ${item.project_count}`} />
                                    </List>
                                </Card.Content>
                            </Card>
                        ))}
                    </Card.Group>
                </div>
            </div>
        )
    }

}