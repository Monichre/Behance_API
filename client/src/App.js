import React, { Component } from 'react'
import { BrowserRouter } from 'react-router-dom'
import {Constants} from './constants.js'
import 'semantic-ui-css/semantic.min.css'
import './App.css'

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        {Constants.routes}
      </BrowserRouter>
    )
  }
}

export default App;
