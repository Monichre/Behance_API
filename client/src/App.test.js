import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import Home from './Components/Home'
import renderer from 'react-test-renderer'


it('renders without crashing', () => {

  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
})


test('Home componentWillMount executes API calls', () => {
  
  const tree = renderer 
    .create(<Home />)
    .toJSON()

  expect(tree).toMatchSnapshot()
  
})
