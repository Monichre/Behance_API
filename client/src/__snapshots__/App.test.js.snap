// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home componentWillMount executes API calls 1`] = `
<div
  id="Home"
>
  <header>
    <i
      className="fa fa-bars"
    />
    <div
      className="menu"
    >
      <div
        className="search-wrapper "
      >
        <div
          className="input-holder"
        >
          <input
            className="search-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Type to search"
            type="text"
            value=""
          />
          <button
            className="search-icon"
            onClick={[Function]}
          >
            <span />
          </button>
        </div>
        <span
          className="close"
          onClick={[Function]}
        />
      </div>
    </div>
  </header>
  <aside>
    <h1>
      Dealer Inspire + Behance
    </h1>
    <ul />
  </aside>
  <div
    className="dashboard"
  >
    <div
      className="search-content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <h1>
        Search Results
      </h1>
      <div
        className="results-column"
      />
    </div>
    <div
      className="ui cards"
    />
  </div>
</div>
`;
